lib/engine/phase.rb

```diff
@@ -100,7 +101,8 @@ def rust_trains!(train, entity)
       end
 
       @game.trains.each do |t|
-        next if t.rusted || t.rusts_on != train.sym
+        next if t.rusted
+        next unless t.rusts_on == train.sym || (t.obsolete_on == train.sym && @depot.discarded.include?(t))
```

Comment:
don't use unless with multiple statements, change this to a next if

Reply:
If I change it to
```
next if !(t.rusts_on == train.sym || (t.obsolete_on == train.sym && @depot.discarded.include?(t)))
```
then rubocop complains.

I could DeMorgan it into
```
 next if t.rusts_on != train.sym && !(t.obsolete_on == train.sym && @depot.discarded.include?(t))
```
but I think that's pretty ugly and actually obfuscates the meaning, which is "skip the following code unless the train should be rusted".

I would be fine with
```
 should_rust = t.rusts_on == train.sym || (t.obsolete_on == train.sym && @depot.discarded.include?(t))
 next unless should_rust
```

But it's your call.